client/post-editor/post-editor.jsx

```diff
@@ -650,14 +649,12 @@ export const PostEditor = React.createClass( {
 
 	onPublish: function() {
 		const edits = {
-			...this.props.edits,
-			status: 'publish'
+			status: 'publish',
+			...this.props.edits
 		};
 
-		// determine if this is a private publish
-		if ( utils.isPrivate( this.state.post ) ) {
-			edits.status = 'private';
-		} else if ( utils.isFutureDated( this.state.post ) ) {
+		// determine if this is a future publish
+		if ( utils.isFutureDated( this.state.post ) ) {
```

Comment:
I'm not sure I understand your comment properly. Anyway, dropping this check has no effect on the behavior of the editor on my testings because even if we save a post with a status "publish" and a future date, the api will sanitize it to "future" as a response